code/game/gamemodes/malfunction/Malf_Modules.dm

```diff
@@ -626,6 +629,100 @@
 		to_chat(src, "<span class='warning'>[alert_msg]</span>")
 	return success
 
+//Turret Assembly: Assemble an AI turret at the chosen location. One use per purchase
+/datum/AI_Module/place_turret
+	module_name = "Deploy Turret"
+	mod_pick_name = "turretdeployer"
+	description = "Build a turret anywhere that lethally targets organic life in sight."
+	cost = 30
+	power_type = /datum/action/innate/ai/place_turret
+	unlock_text = "<span class='notice'>You prepare an energy turret for deployment.</span>"
+	unlock_sound = 'sound/items/rped.ogg'
+
+/datum/action/innate/ai/place_turret
+	name = "Deploy Turret"
+	desc = "Build a turret anywhere that lethally targets organic life in sight."
+	button_icon_state = "deploy_turret"
+	uses = 1
+	auto_use_uses = FALSE
+	var/image/turfOverlay
+
+/datum/action/innate/ai/place_turret/New()
+	..()
+	turfOverlay = image("icon"='icons/turf/overlays.dmi')
+
+/datum/action/innate/ai/place_turret/Activate()
+	if(active)
+		to_chat(owner, "<span class='notice'>Your assemblers can only construct one turret at a time.</span>")
+		return
+	if(!owner_AI.can_place_turret(src))
+		return
+	active = TRUE
+	if(alert(owner, "Are you sure you want to place a turret here? Deployment will take a few seconds to complete, in which the turret will be vulnerable.", "Are you sure?", "Yes", "No") == "No")
```

Comment:
```suggestion
	if(alert(owner, "Are you sure you want to place a turret here? Deployment will take a few seconds to complete, in which the turret will be vulnerable.", "Are you sure?", "Yes", "No") == "No")
```
Uh, don't think this needs an alert, just use do_after once

Reply:
It should probably throw this alert to prevent silly misclicks from happening. 
If what you mean is that with `do_after_once` you can activate the ability again to stop it, that seems slightly counterintuitive to me

Reply:
This alert is to prevent misclicks, as this ability if used prematurely or in an inopportune location can range from being a waste of CPU to being actively detrimental.

Reply:
> This alert is to prevent misclicks, as this ability if used prematurely or in an inopportune location can range from being a waste of CPU to being actively detrimental.

This probably won't work because it autos over yes and you'd be spamming yes anyways so they'd just jam enter before their brain catches up, just have it be based on a tile you click and then have it be a do after once

Reply:
I don't see that in the tests - the yes is auto-selected if you're using enter, but once it starts deploying you shouldn't be able to halt it. This is also how it works with the borg factory, in that you get an alert that asks if you're sure before it deploys.

Reply:
The alert now has the default selection as No. It already defaulted to no if the window closed. This should handle preventing accidental deployments.